File: typeScript/challenges/challenge08.ts
export function drawRace (indices: number[], length: number): string {
  const lane: string[] = '~'.repeat(length).split('')
  const lengthIndeces: number = indices.length

  const raceTrackArr: string[] = indices.map((indice, index) => {
    const wideRight: string = ' '.repeat(lengthIndeces - (index + 1))
    if (indice === 0) return `${wideRight}${lane.join('')} /${index + 1}`

    const laneReindeer: string = lane.toSpliced(indice, 1, 'r').join('')
    return `${wideRight}${laneReindeer} /${index + 1}`
  })
  const raceTrack: string = raceTrackArr.join('\n')
  return raceTrack
}

File: python/challenge08.py
def draw_race(indices, length):
    race_text = []
    for i,j in enumerate(indices):
        if j == -1:
            race_text.append('~'*length)
            race_text[i] = race_text[i][:j] + 'r'
            race_text[i] = ' '*((len(indices)-1)-i) + race_text[i]
        elif j != 0:
            race_text.append('~'*length)
            race_text[i] = race_text[i][:j] + 'r' + race_text[i][j + 1:]
            race_text[i] = ' '*((len(indices)-1)-i) + race_text[i]
        else:
            race_text.append('~'*length)
            race_text[i] = ' '*((len(indices)-1)-i) + race_text[i]
        
        race_text[i] += f" /{i+1}"

    return '\n'.join(race_text)


race = [2, -1, 0, 5]
lens = 10

print(draw_race(race,lens))



